pkg/dhclient/dhclient.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2013-2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package dhclient
+
+import (
+	"context"
+	"log"
+	"regexp"
+	"time"
+
+	"github.com/u-root/u-root/pkg/dhclient"
+	"github.com/vishvananda/netlink"
+)
+
+// Request sets up the dhcp configurations for all of the ifNames.
+func Request(ifName string, timeout int, retry int, verbose bool, ipv4 bool, ipv6 bool) {
+	ifRE := regexp.MustCompilePOSIX(ifName)
+
+	ifnames, err := netlink.LinkList()
+	if err != nil {
+		log.Fatalf("Can't get list of link names: %v", err)
+	}
+
+	var filteredIfs []netlink.Link
+	for _, iface := range ifnames {
+		if ifRE.MatchString(iface.Attrs().Name) {
+			filteredIfs = append(filteredIfs, iface)
+		}
+	}
+
+	if len(filteredIfs) == 0 {
+		log.Fatalf("No interfaces match %s", ifName)
+	}
+
+	channel := make(chan error)
+	go configureAll(filteredIfs, channel, timeout, retry, verbose, ipv4, ipv6)
+	x := <-channel
+	log.Printf("The channel returns: %v", x)
+}
+
+func configureAll(ifs []netlink.Link, channel chan error, timeout int, retry int, verbose bool, ipv4 bool, ipv6 bool) {
```

Comment:
Give the channel parameter a better name, something like "results".

Reply:
The type can also be the send-only channel type "chan<- error".

Reply:
That is a good idea. As of right now, the channel could receive information to it. It is more type safety. With this, I can only send errors up. Thanks for the catch :D